src/meta/src/rpc/server.rs

```diff
@@ -171,316 +153,312 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
-    if let Some(ref dashboard_addr) = address_info.dashboard_addr {
-        let dashboard_service = crate::dashboard::DashboardService {
-            dashboard_addr: *dashboard_addr,
-            cluster_manager: cluster_manager.clone(),
-            fragment_manager: fragment_manager.clone(),
-            meta_store: env.meta_store_ref(),
-            prometheus_endpoint: prometheus_endpoint.clone(),
-            prometheus_client: prometheus_endpoint.as_ref().map(|x| {
-                use std::str::FromStr;
-                prometheus_http_query::Client::from_str(x).unwrap()
-            }),
-        };
-        // TODO: join dashboard service back to local thread.
-        tokio::spawn(dashboard_service.serve(address_info.ui_path));
+    /// Check for `number_of_nodes` meta leader nodes, starting at `meta_port`, `meta_port + 1`, ...
+    /// Simulates `number_of_nodes` compute nodes, starting at `meta_port`, `meta_port + 1`, ...
+    ///
+    /// ## Returns
+    /// Number of nodes which currently are leaders. Number is not snapshoted. If there is a
+    /// leader failover in process, you may get an incorrect result
+    async fn _number_of_leaders(number_of_nodes: u16, meta_port: u16, host_port: u16) -> u16 {
+        let mut leader_count = 0;
+        for i in 0..number_of_nodes {
+            let local = "127.0.0.1".to_owned();
+            let port = meta_port + i;
+            let meta_addr = format!("http://{}:{}", local, port);
+            let host_addr = HostAddr {
+                host: local,
+                port: host_port + i,
+            };
+
+            let endpoint = Endpoint::from_shared(meta_addr.to_string())
+                .unwrap()
+                .initial_connection_window_size(MAX_CONNECTION_WINDOW_SIZE);
+            let channel = endpoint
+                .http2_keep_alive_interval(Duration::from_secs(60))
+                .keep_alive_timeout(Duration::from_secs(60))
+                .connect_timeout(Duration::from_secs(5))
+                .connect()
+                .await
+                .inspect_err(|e| {
+                    tracing::warn!(
+                        "Failed to connect to meta server {}, wait for online: {}",
+                        meta_addr,
+                        e
+                    );
+                })
+                .unwrap();
+
+            let cluster_client = ClusterServiceClient::new(channel);
+            let resp = cluster_client
+                .to_owned()
+                .add_worker_node(AddWorkerNodeRequest {
+                    worker_type: WorkerType::ComputeNode as i32,
+                    host: Some(HostAddress {
+                        host: host_addr.host,
+                        port: host_addr.port as i32,
+                    }),
+                    worker_node_parallelism: 5,
+                })
+                .await;
+            let is_leader = resp.is_ok();
+
+            if is_leader {
+                leader_count += 1;
+            }
```

Comment:
```suggestion
            if resp.is_ok() {
                leader_count += 1;
            }
```
Nit.

Reply:
Let me add a comment, so that future devs know what we are counting here 😁